### PR TITLE
Added WPS monitoring

### PR DESCRIPTION
**Pull request by bhasin85**

### CODE REVIEW

File: web/src/main/java/org/fao/geonet/monitor/onlineresource/OnlineResourceCheckerWps.java [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+            connection = (HttpURLConnection) (new URL(url)).openConnection();
+            connection.setConnectTimeout(OnlineResourceMonitorService.timeout * 1000);
+            connection.setReadTimeout(OnlineResourceMonitorService.timeout * 1000);
+            connection.setRequestMethod("POST");
+            connection.setRequestProperty("Content-Type", "text/xml");
+            connection.setDoOutput(true);
+
+            OutputStreamWriter writer = null;
+            try {
+                writer = new OutputStreamWriter(connection.getOutputStream());
```
Comment (julian1) [inline web/src/main/java/org/fao/geonet/monitor/onlineresource/OnlineResourceCheckerWps.java]: Nice resource handling!

File: web/src/main/java/org/fao/geonet/monitor/onlineresource/OnlineResourceCheckerWps.java [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+                "    <wps:Input>\n" +
+                "      <ows:Identifier>layer</ows:Identifier>\n" +
+                "      <wps:Data>\n" +
+                "        <wps:LiteralData>" + layer + "</wps:LiteralData>\n" +
+                "      </wps:Data>\n" +
+                "    </wps:Input>\n" +
+                "    <wps:Input>\n" +
+                "      <ows:Identifier>subset</ows:Identifier>\n" +
+                "      <wps:Data>\n" +
+                "        <wps:LiteralData>TIME," + literalTime + "," + literalTime + ";LATITUDE,-31.6855,-31.6855;LONGITUDE,114.8291,114.8291</wps:LiteralData>\n" +
```
Comment (julian1) [inline web/src/main/java/org/fao/geonet/monitor/onlineresource/OnlineResourceCheckerWps.java]: Are we sure that there will always be data for gogoduck requests in this spatial extent?

File: web/src/main/java/org/fao/geonet/monitor/onlineresource/OnlineResourceCheckerWps.java [hunk 3]
```diff
@@ -0,0 +1,179 @@
...
+            if (connection.getContentType().equals("text/csv;charset=UTF-8")) {
+                InputStream is = null;
+                CSVReader reader;
+                try {
+                    is = connection.getInputStream();
+                    reader = new CSVReader(new InputStreamReader(connection.getInputStream()));
+                    String[] line;
+
+                    int lineCount = 1;
+                    while ((line = reader.readNext()) != null && line.length == 5) {
```
Comment (julian1) [inline web/src/main/java/org/fao/geonet/monitor/onlineresource/OnlineResourceCheckerWps.java]: Why 5?

File: web/src/main/java/org/fao/geonet/monitor/onlineresource/OnlineResourceCheckerWps.java [hunk 4]
```diff
@@ -0,0 +1,179 @@
...
+                        if (lineCount == 2) {
+                            time = line[4];
```
Comment (julian1) [inline web/src/main/java/org/fao/geonet/monitor/onlineresource/OnlineResourceCheckerWps.java]: And the forth column? Is this consistent for all gogoduck requests. Would it be better to use variables that explain what is going on. Eg. expectedTimeColumn ?

File: web/src/main/java/org/fao/geonet/monitor/onlineresource/OnlineResourceCheckerWps.java (hunk 3 above)
Comment (bhasin85) [inline web/src/main/java/org/fao/geonet/monitor/onlineresource/OnlineResourceCheckerWps.java]: Updated the logic

File: web/src/main/java/org/fao/geonet/monitor/onlineresource/OnlineResourceCheckerWps.java (hunk 4 above)
Comment (bhasin85) [inline web/src/main/java/org/fao/geonet/monitor/onlineresource/OnlineResourceCheckerWps.java]: Updated the logic

File: web/src/main/java/org/fao/geonet/monitor/onlineresource/OnlineResourceCheckerWps.java (hunk 2 above)
Comment (bhasin85) [inline web/src/main/java/org/fao/geonet/monitor/onlineresource/OnlineResourceCheckerWps.java]: We make WFS request to get the time. So I think we are pretty sure that there will be some data. If not the check will fail anyway.

**Comment by julian1** [comment]
Agree with your point that we need to refactor the duplicated connection stuff with the other checkers etc.

**Comment by julian1** [comment]
Something for later..

**Pull request merged**